adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd.features
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+
+/**
+ * RDD holding run length encoded coverage in a FeatureRDD. CoverageRDD stores
+ * coverage counts at a ReferenceRegion in the score attribute of
+ * a feature.
+ *
+ * @param featureRdd FeatureRDD holding records for coverage and corresponding
+ * referenceRegion
+ */
+case class CoverageRDD(featureRdd: FeatureRDD) {
+
+  /**
+   * Saves coverage as feature file.
+   * @see FeatureRDD.save
+   *
+   * Supported file formats include bed, narrowPeak and parquet. Coverage is saved
+   * as a feature where coverage is stored in score attribute.
+   *   val chrom    = feature.getContigName
+   *   val start    = feature.getStart
+   *   val end      = feature.getEnd
+   *   val name     = Features.nameOf(feature)
+   *   val coverage = feature.getScore
+   *
+   * @param filePath The location to write the output.
+   */
+  def save(filePath: java.lang.String) = {
+    featureRdd.save(filePath)
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is coverage of the first base pair in that bin.
+   *
+   * @param region ReferenceRegion to fetch overlapping coverage from
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def getCoverage(region: ReferenceRegion, bpPerBin: Int = 1): RDD[Coverage] = {
+
+    val flattenedCoverage = features2Coverage(region)
+
+    if (bpPerBin == 1) {
+      flattenedCoverage // no binning, return raw results
+    } else {
+      // subtract region.start to shift mod to start of ReferenceRegion
+      flattenedCoverage.filter(r => (r.start - region.start) % bpPerBin == 0)
+    }
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is the mean coverage over all base pairs in that bin.
+   *
+   * @param region ReferenceRegion to fetch overlapping coverage from
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def getAggregatedCoverage(region: ReferenceRegion, bpPerBin: Int = 1): RDD[Coverage] = {
+
+    val flattenedCoverage = features2Coverage(region)
+
+    if (bpPerBin == 1) {
+      flattenedCoverage // no binning, return raw results
+    } else {
+      flattenedCoverage.map(r => {
+        // key coverage by binning start site mod bpPerbin
+        // subtract region.start to shift mod to start of ReferenceRegion
+        val key = r.start - ((r.start - region.start) % bpPerBin)
+        (key, r)
+      }).groupByKey
```

Comment:
How could we fit getCoverage to filterByOverlappingRegion? I beleive these are different functions because filterByOverlappingRegion returns raw records. For example, if the coverage records are not merged to begin with, then filterByOverlappingRegion will return chunks of coverage. Also, filterByOverlappingRegion cannot specify binning.